.github/workflows/ql-for-ql-build.yml

```diff
@@ -0,0 +1,160 @@
+name: Run QL for QL
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  queries:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache queries
+        id: cache-queries
+        uses: actions/cache@v2
+        with:
+          path: ${{ runner.temp }}/query-pack.zip
+          key: ${{ runner.os }}-queries-${{ hashFiles('**/*.ql*') }}
+      - name: Find codeql
+        if: steps.cache-queries.outputs.cache-hit != 'true'
+        id: find-codeql
+        uses: github/codeql-action/init@esbena/ql
+        with:
+          languages: javascript # does not matter
```

Comment:
```suggestion
          languages: ql # does not matter
```
sligthly nicer

Reply:
Nope, doesn't work, I'm not sure why.

Reply:
Can you link the failure here?

Reply:
I think it's this one: https://github.com/github/codeql-ql/runs/4534309639?check_suite_focus=true

Reply:
ql-code-scanning.qls 😂 if only

Reply:
Oh, wait. That file is actually there 🤔

Reply:
The reason it does not work here, but later, is that the default query suite name does not exist yet. It exists once we do the proper setup in the context of our codeql-cli-with-ql.

Reply:
So it is just a bootstrapping problem. Lets just stick to `javascript` then.

Reply:
We don't really need the `codeql-action` at all and simply use the `codeql` CLI directly. We shouldn't even need the CodeQL bundle, a plain CLI should be enough.

Reply:
I'm abusing the codeql-action here to tell me where to find the codeql cli on disk.

Reply:
> I'm abusing the codeql-action here to tell me where to find the codeql cli on disk.

Yes, I figured that. I think we could rewrite the workflows to use `gh codeql` instead and reduce the amount of (ab)use of the codeql-action ;-) Best left for a follow-up pull request though.

Reply:
> Yes, I figured that. I think we could rewrite the workflows to use gh codeql instead and reduce the amount of (ab)use of the codeql-action ;-) Best left for a follow-up pull request though. 

`codeql-action` uses a CodeQL installation that is cached on all the workers, and therefore we don't have to download a fresh CodeQL installation each time.  
So yes, we are (ab)using `codeql-action`, but I don't think we are going to change that.

Reply:
@erik-krogh Is that to "speed things up". The CodeQL Action Init step takes about 9 seconds and fetching a fresh CodeQL CLI takes 13 seconds. Using CodeQL from the tool-cache is indeed faster, but not by much .